odoo/fields.py

```diff
@@ -964,7 +964,7 @@ def __get__(self, record, owner):
                     ]))
                 value = env.cache.get(record, self)
 
-            elif self.store and record._origin:
+            elif self.store and record._origin and self.compute and not self.readonly:
                 # new record with origin: fetch from origin
                 value = self.convert_to_cache(record._origin[self.name], record)
                 env.cache.set(record, self, value)
```

Comment:
Suppose `balance` that is a computed stored field `debit + credit = balance` not inside the view. Since this field is not `readonly`, it should be recomputed instead of picking its value to the `_origin`.